### PR TITLE
jsonnet: fix rollout-operator ports

### LINKED ISSUE #10272: Bug: jsonnet: rollout-operator: Wrong ports in container and service
### What is the bug?

With generation of manifests and enabled rollout-operator and enable_rollout_operator_webhook=true . 
Deployment has one open port 8001. In this case, service uses port 8443

<details>
  <summary>apps-v1.Deployment-rollout-operator.yaml</summary>

```yaml
 apiVersion: apps/v1
kind: Deployment
metadata:
  labels:
    tanka.dev/environment: 47e6facd808ae73b80fbfae3c2e8c175348504d7f30ef622
  name: rollout-operator
  namespace: mimir
spec:
  minReadySeconds: 10
  replicas: 1
  revisionHistoryLimit: 10
  selector:
    matchLabels:
      name: rollout-operator
  strategy:
    rollingUpdate:
      maxSurge: 0
      maxUnavailable: 1
  template:
    metadata:
      labels:
        name: rollout-operator
    spec:
      containers:
      - args:
        - --server-tls.enabled=true
        - -kubernetes.namespace=mimir
        image: grafana/rollout-operator:v0.19.1
        imagePullPolicy: IfNotPresent
        name: rollout-operator
        ports:
        - containerPort: 8001
          name: http-metrics
        readinessProbe:
          httpGet:
            path: /ready
            port: 8001
          initialDelaySeconds: 5
          timeoutSeconds: 1
        resources:
          limits:
            cpu: 200m
            memory: 200Mi
          requests:
            cpu: 100m
            memory: 200Mi
      serviceAccountName: rollout-operator
```
</details>

<details>
  <summary>v1.Service-rollout-operator.yaml</summary>

```yaml
apiVersion: v1
kind: Service
metadata:
  labels:
    tanka.dev/environment: 47e6facd808ae73b80fbfae3c2e8c175348504d7f30ef622
  name: rollout-operator
  namespace: mimir
spec:
  ports:
  - name: https
    port: 443
    protocol: TCP
    targetPort: 8443
  selector:
    name: rollout-operator

```
</details>

### How to reproduce it?

1. Install the mimir project
  ```
  tk init --k8s=1.29
  # Install Mimir jsonnet.
  jb install github.com/grafana/mimir/operations/mimir@mimir-2.14.2
  ```
2. Create file in directory environments/default/main.jsonnet , with content:
<details>
  <summary>environments/default/main.jsonnet</summary>

```
local mimir = import "mimir/mimir.libsonnet";

mimir {
  _config+:: {
    deployment_mode: 'microservices',
    namespace: 'mimir',
    cluster: 'mimir',
    etcd_replicas: 0,
    limits: $._config.overrides.medium_user,
    overrides+: {
      // TODO: может не medium user
      medium_user+: {
        ruler_max_rules_per_rule_group: 50,
        compactor_blocks_retention_period: '90d',
        query_sharding_total_shards: 8,
      },
    },

    multi_zone_ingester_enabled: true,
    multi_zone_ingester_replicas: 3,
    multi_zone_store_gateway_enabled: true,
    multi_zone_store_gateway_replicas: 3,
    memberlist_ring_enabled: true,
    enable_rollout_operator_webhook: true,
    ingester_automated_downscale_v2_enabled: true,
    consul: null,
    etcd: null,
    storage_backend: 'gcs',

    blocks_storage_bucket_name: 'example-bucket',
    node_selector: {
      workload: 'mimir',
    },
  },
}

```
</details>

3. Export manifests: `tk export default  environments/default/`
4. Check it out v1.Service-rollout-operator.yaml and v1.Service-rollout-operator.yaml

### What did you think would happen?

The service and deployment must have the same set of ports 8001 and 8443

### What was your environment?

Infrastructure: kubernetes  
Deployment tool: jsonnet
Version: mimir-2.14.2

### Any additional context to share?

_No response_

### PR DESCRIPTION
**Pull request by sentoz**
#### What this PR does

Implements correct port specification for rollout-operator in deployment, service, MutatingWebhookConfiguration and ValidatingWebhookConfiguration

#### Which issue(s) this PR fixes or relates to

Fixes #10272 

#### Checklist

- [x] Tests updated.
- [x] `CHANGELOG.md` updated - the order of entries should be `[CHANGE]`, `[FEATURE]`, `[ENHANCEMENT]`, `[BUGFIX]`.

### CODE REVIEW

File: operations/mimir/rollout-operator.libsonnet [hunk 1]
```diff
@@ -53,9 +53,14 @@
   rollout_operator_container::
     container.new('rollout-operator', $._images.rollout_operator) +
     container.withArgsMixin($.util.mapToFlags($.rollout_operator_args)) +
-    container.withPorts([
-      $.core.v1.containerPort.new('http-metrics', 8001),
-    ]) +
+    container.withPorts(
```
Comment (56quarters) [inline operations/mimir/rollout-operator.libsonnet:56]: This change makes sense to me but can be rewritten to something like:

```
container.withPorts(
  [$.core.v1.containerPort.new('http-metrics', 8001)] + if $._config.enable_rollout_operator_webhook then [$.core.v1.containerPort.new('https', 8443)] else []
)
```

File: operations/mimir/rollout-operator.libsonnet [hunk 2]
```diff
@@ -196,7 +196,7 @@
             name: 'rollout-operator',
             namespace: $._config.namespace,
             path: '/admission/no-downscale',
-            port: 443,
+            port: 8443,
```
Comment (56quarters) [inline operations/mimir/rollout-operator.libsonnet:199]: We don't want to change this port.

File: operations/mimir/rollout-operator.libsonnet [hunk 3]
```diff
@@ -74,12 +79,7 @@
     $.newMimirNodeAffinityMatchers($.rollout_operator_node_affinity_matchers),
 
   rollout_operator_service: if !rollout_operator_enabled || !$._config.enable_rollout_operator_webhook then null else
-    service.new(
-      'rollout-operator',
-      { name: 'rollout-operator' },
-      servicePort.newNamed('https', 443, 8443) +
-      servicePort.withProtocol('TCP'),
-    ),
+    $.util.serviceFor($.rollout_operator_deployment, $._config.service_ignored_labels),
```
Comment (56quarters) [inline operations/mimir/rollout-operator.libsonnet:82]: We don't want to change the HTTPS port of the service.

**Review by 56quarters — CHANGES_REQUESTED** [review]
The issue #10272 didn't mention _why_ you want the service and containers to use the same port for HTTPS. The current configuration works today and so we don't want to change this.

File: operations/mimir/rollout-operator.libsonnet (hunk 1 above)
Comment (sentoz) [reply operations/mimir/rollout-operator.libsonnet:56]: I agree it looks better.

**Comment by sentoz** [comment]
> The issue #10272 didn't mention _why_ you want the service and containers to use the same port for HTTPS. The current configuration works today and so we don't want to change this.

Using the same service and container port is not essential. The main problem is that only port 8001 was declared in the container, and only port 8443/443 was declared in the service. It turns out that the call from webhooks is unreachable. 

Changing the ports is just a reduction to a unified approach to creating services (returned to the original one, adding another port).

**Review by 56quarters — APPROVED** [review]
Thanks!

**Pull request merged**